src/renderers/shared/ReactDebugTool.js

```diff
@@ -279,26 +299,78 @@ var ReactDebugTool = {
     checkDebugID(debugID);
     checkDebugID(parentDebugID, true);
     emitEvent('onBeforeMountComponent', debugID, element, parentDebugID);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var isCompositeComponent = typeof element.type === 'function';
+      if (isCompositeComponent) {
+        var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+        var label = displayName + ' (mount)';
+        console.time(label);
+      }
+    }
   },
   onMountComponent(debugID) {
     checkDebugID(debugID);
     emitEvent('onMountComponent', debugID);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var element = ReactComponentTreeHook.getElement(debugID);
+      var isCompositeComponent = typeof element.type === 'function';
+      if (isCompositeComponent) {
+        var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+        var label = displayName + ' (mount)';
+        console.timeEnd(label);
+      }
+    }
   },
   onBeforeUpdateComponent(debugID, element) {
     checkDebugID(debugID);
     emitEvent('onBeforeUpdateComponent', debugID, element);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var isCompositeComponent = typeof element.type === 'function';
+      if (isCompositeComponent) {
+        var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+        var label = displayName + ' (update)';
+        console.time(label);
+      }
+    }
   },
   onUpdateComponent(debugID) {
     checkDebugID(debugID);
     emitEvent('onUpdateComponent', debugID);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var element = ReactComponentTreeHook.getElement(debugID);
+      var isCompositeComponent = typeof element.type === 'function';
```

Comment:
Not really about your diff but should we have a unique way to check if an element is composite? ReactTestUtil has a different way of testing it: https://github.com/facebook/react/blob/1c5a639c37039a9c60752980dfee168129b76b32/src/test/ReactTestUtils.js#L114-L123